### ISSUE TITLE
Selecting single spec from spec list causes all tests to rerun 

### ISSUE THREAD

**Issue by edge33** [karma-runner/karma#3581]
#### Issue description

running `ng test` in angular project and selecting single spec from spec list causes all tests to rerun 

#### Steps to reproduce the issue

1.  npm install -g @angular/cli
2.  ng new test-app
3. cd test-app
4. ng test
5. clic on a single spec

#### What's the expected result?

- only selected spec gets to run again

#### What's the actual result?

- all the specs are re run

#### Additional details / screenshot

- Dependencies:

```
"@angular-devkit/build-angular": "~0.1001.4",
"@angular/cli": "~10.1.4",
"@angular/compiler-cli": "~10.1.4",
"@types/node": "^12.11.1",
"@types/jasmine": "~3.5.0",
"@types/jasminewd2": "~2.0.3",
"codelyzer": "^6.0.0",
"jasmine-core": "~3.6.0",
"jasmine-spec-reporter": "~5.0.0",
"karma": "~5.0.0",
"karma-chrome-launcher": "~3.1.0",
"karma-coverage-istanbul-reporter": "~3.0.2",
"karma-jasmine": "~4.0.0",
"karma-jasmine-html-reporter": "^1.5.0",
"protractor": "~7.0.0",
"ts-node": "~8.3.0",
"tslint": "~6.1.0",
"typescript": "~4.0.2"
```

**Comment by johnjbarton** [comment]
Specs and spec selection is done by plugins, not by the core code. The core code has no idea what a "spec" even is ;-)

**Comment by philno** [comment]
Hi guys, 
I was running into the same issue with angular versions 9 and 11. I updated all karma/jasmine related packages to the latest version:

```
    "jasmine-core": "~3.6.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~5.2.3",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
```

Using Windows 10 / Chrome Version 87.0.4280.88.

When clicking on a single spec the whole page reloads and all tests are re-run.
Somewhere it was mentioned that using the debug mode helps. However, clicking a spec in debug mode also redirects me to the regular karma page and thus reruns all tests.

I found a workaround by changing the `toQueryString` function to always include `debug.html`.
To apply the fix: Open `node_modules\karma-jasmine-html-reporter\src\lib\html.jasmine.reporter.js` and look for the `toQueryString` function at the bottom of the file. 
Change the following line: 
```JavaScript
return '?' + qStrPairs.join('&');
``` 
to 
```JavaScript
return 'debug.html?' + qStrPairs.join('&');
``` 
For `karma-jasmine-html-reporter@1.5.4`, this is in line 794.
This leads to the expected behaviour: Clicking on a component or spec just reruns the tests of that component or spec.

I'm not sure why this is happening with different angular / karma versions. Maybe someone can clear up which package is causing this problem. I hope that this problem can be fixed so that we don't have to rely on a workaround.
